### ISSUE TITLE
Add option to strip diacritical marks

### ISSUE THREAD

**Issue by ing-fco-rguez** [ericgio/react-bootstrap-typeahead#124]

**Comment by ing-fco-rguez** [comment]
Hi, first of all let me thank you for your awsome job on this component.

My issue is that I'm still not able to figure out how to use characters without accents in the filter of options with accents, my ajax call for data (options) does accepts this and returns information, but when I type a character without accent it displays the "no results found" message.

Thank you very much in advance.

**Comment by ericgio** [comment]
Thanks for the report. The current filter is case-insensitive, but does match diacritical marks. I'd need to add an option to the filter to ignore those.

**Comment by ericgio** [comment]
As a temporary workaround, you could create a custom filter with the `filterBy` prop and [use a solution like this](http://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript/18391901#18391901).

**Comment by ing-fco-rguez** [comment]
Thank you very much for your response, I was actually doing some tests whit the `filterBy` option and with your advice, my function now looks like this:

```
filterBy = {option => {
	removeDiacritics(option['place'].toLowerCase()).indexOf(removeDiacritics(this.state.text.toLowerCase())) > 0;
}}
```

The condition is indeed true for some of the options but none is rendering, only the "no matches found" text is displayed.

Thanks a lot in advance for your help.

**Comment by ericgio** [comment]
Looks like you need `>= 0` or `!== -1` otherwise it will return false if you type in a string that matches the beginning of the label.

**Comment by ing-fco-rguez** [comment]
Thank you for your response, but I've tried both conditions, still no options are displayed :(

```
filterBy = {option => {
	console.log('[header-search-box.js][Typeahead] - Status: ', removeDiacritics(option['place'].toLowerCase()).indexOf(removeDiacritics(this.state.text.toLowerCase())) >= 0 ? 'Found' : 'Not found');
	removeDiacritics(option['place'].toLowerCase()).indexOf(removeDiacritics(this.state.text.toLowerCase())) >= 0;
}}

```
In the log I can see several "Found" and some "Not found" being printed, but nothing the "no matches found" message remains.

**Comment by ing-fco-rguez** [comment]
Done, I was missing a 'return' there (u_u)

```
filterBy = {option => {
return removeDiacritics(option['place'].toLowerCase()).indexOf(removeDiacritics(this.state.text.toLowerCase())) >= 0;
}}
```

Thanks a lot 👍

**Comment by ericgio** [comment]
I'm going to keep this open for now, as a task to add this functionality.

**Comment by f4k3N4m3** [comment]
Hi ericgio, Create Job!
Please add 'markBy' props so we can choose what to mark as bold after the filterBy... thanks!

**Comment by f4k3N4m3** [comment]
Hi ericgio, great Job, thanks for Awesome component!
Please add 'markBy' props so we can choose what to mark as bold after the filterBy... thanks!

**Comment by ericgio** [comment]
@f4k3N4m3: Can you provide an example? What would you want to mark as bold besides what the user has input? By the way, you can already control this by using `renderMenuItemChildren` to render the contents of the menu items exactly as you want.

**Comment by f4k3N4m3** [comment]
@ericgio  Hi, sorry for late answer. 
I want to ignore some characters but still bold the relevant results.
for example if we have:
de'longhi
model

but the user search for **del**
he will see
**de'l**onghi
mo**del**.

(I did it with filterBy...)

I'll check renderMenuItemChildren if it's suit. 
Thanks again!

**Comment by f4k3N4m3** [comment]
@ericgio renderMenuItemChildren worked perfectly,
Thanks!

**Comment by ericgio** [comment]
`ignoreDiacritics` now available in v1.0.0
